pwndbg/commands/branch.py

```diff
@@ -0,0 +1,111 @@
+import argparse
+
+import gdb
+from capstone import CS_GRP_JUMP
+
+import pwndbg.color.message as message
+import pwndbg.commands
+import pwndbg.disasm
+import pwndbg.gdblib.bpoint
+import pwndbg.gdblib.next
+
+
+class BreakOnConditionalBranch(pwndbg.gdblib.bpoint.Breakpoint):
+    """
+    A breakpoint that only stops the inferior if a given branch is taken or not taken.
+    """
+
+    def __init__(self, instruction, taken):
+        super().__init__("*%#x" % instruction.address, type=gdb.BP_BREAKPOINT, internal=False)
+        self.instruction = instruction
+        self.taken = taken
+
+    def should_stop(self):
+        # Use the assistant to figure out which if all the conditions this
+        # branch requires in order to be taken have been met.
+        assistant = pwndbg.disasm.arch.DisassemblyAssistant.for_current_arch()
+        condition_met = assistant.condition(self.instruction)
+        if condition_met is None:
+            # This branch is unconditional.
+            condition_met = 1
+        condition_met = condition_met != 0
+
+        return condition_met == self.taken
+
+
+parser = argparse.ArgumentParser(description="Breaks on a branch if it is taken.")
+parser.add_argument(
+    "branch",
+    type=str,
+    help="The branch instruction to break on.",
+)
+
+
+@pwndbg.commands.ArgparsedCommand(parser, command_name="break-if-taken")
+@pwndbg.commands.OnlyWhenRunning
+def break_if_taken(branch) -> None:
+    """Breaks at the next call instruction"""
+    install_breakpoint(branch, taken=True)
+
+
+parser = argparse.ArgumentParser(description="Breaks on a branch if it is not taken.")
+parser.add_argument(
+    "branch",
+    type=str,
+    help="The branch instruction to break on.",
+)
+
+
+@pwndbg.commands.ArgparsedCommand(parser, command_name="break-if-not-taken")
+@pwndbg.commands.OnlyWhenRunning
+def break_if_not_taken(branch) -> None:
+    """Breaks at the next call instruction"""
+    install_breakpoint(branch, taken=False)
+
+
+def install_breakpoint(branch, taken):
+    # Do our best to interpret branch as a locspec. Untimately, though, we're
+    # limited in what we can do from inside Python in that front.
+    address = None
+    try:
+        # Try to interpret branch as an address literal
+        address = int(branch, 0)
+    except ValueError:
+        pass
+
+    if address is None:
```

Comment:
Should this just be in the except branch maybe?

Reply:
Yeah, that makes more sense